andreeapetcu/assignment4/CountingIslands.java

```diff
@@ -0,0 +1,285 @@
+/**
+ * Contains two different approaches to solving the counting islands assignment:
+ * 1) countingIslands - non-recursive algorithm, using a disjoint set data structure
+ * 2) countIslands - recursive flood fill algorithm
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CountingIslands {
+  private static final boolean WATER = false;
+
+  /**
+   * Counts the number of islands on a given map.
+   *
+   * NON-RECURSIVE ALGORITHM - main method
+   *
+   * @param rows    the number of rows in the map
+   * @param columns the number of columns in the map
+   * @param map     a two-dimensional map of tiles
```

Comment:
It might be worth saying here what true means (land) and what false means (water)